src/main/java/graphql/execution/incremental/IncrementalCallState.java

```diff
@@ -85,13 +87,19 @@ public boolean getIncrementalCallsDetected() {
         return incrementalCallsDetected.get();
     }
 
+    private Supplier<SingleSubscriberPublisher<DelayedIncrementalPartialResult>> createPublisher() {
+        // this will be created once and once only - any extra calls to .get() will return the previously created
+        // singleton object
+        return new InterThreadMemoizedSupplier<>(() -> new SingleSubscriberPublisher<>(this::drainIncrementalCalls));
+    }
+
     /**
-     * When this is called the deferred execution will begin
+     * When this is called the deferred execution will begin once the {@link Publisher} is subscribed
+     * to.
      *
      * @return the publisher of deferred results
      */
     public Publisher<DelayedIncrementalPartialResult> startDeferredCalls() {
-        drainIncrementalCalls();
```

Comment:
before this would start the deferred processing eagerly.  It would begin to drain the deferred work before returning the publisher